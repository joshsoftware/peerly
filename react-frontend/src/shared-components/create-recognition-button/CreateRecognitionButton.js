import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import PlusSign from "shared-components/plus-sign/PlusSign"; //todo depend upon peerly logo

const WrapperCircle = styled.div`
  background: var(--white) 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px var(--box-shadow-color);
  border-radius: 50%;
  width: 60px;
  height: 60px;
  position: relative;
`;

const ButtonComponent = styled.div`
  background: transparent url(${(props) => props.imageUrl})
  97% 50% no-repeat
  padding-box;
  box-shadow: 0px 3px 6px #0000004e;
  width: 45px;
  height: 45px;
  background-color: var(--rust);
  border: none;
  border-radius: 50%;
  ”
`;

const CreateRecognitionButton = ({ onClick, imageUrl, theme, fontSize }) => {
  return (
    <WrapperCircle className="m-auto justify-content-center">
      <ButtonComponent
        id="create-recognition-button"
        imageUrl={imageUrl}
        className="m-auto text-center"
        onClick={onClick}
      >
        <PlusSign fontSize={fontSize} theme={theme} />
      </ButtonComponent>
    </WrapperCircle>
  );
};

CreateRecognitionButton.defaultProps = {
  theme: "light",
  fontSize: "27px",
};

CreateRecognitionButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  imageUrl: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(["dark", "light"]),
  fontSize: PropTypes.string,
};

export default React.memo(CreateRecognitionButton);
