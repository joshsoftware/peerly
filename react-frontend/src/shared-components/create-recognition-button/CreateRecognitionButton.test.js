import React from "react";
import { render } from "@testing-library/react";

import CreateRecognitionButton from "shared-components/create-recognition-button/CreateRecognitionButton";

it("create recognition component should render correctly", () => {
  const onClick = () => {
    return true;
  };

  const { asFragment } = render(
    <CreateRecognitionButton onClick={onClick} imageUrl="ccc" theme="dark" />
  );

  expect(asFragment()).toMatchSnapshot();
});
