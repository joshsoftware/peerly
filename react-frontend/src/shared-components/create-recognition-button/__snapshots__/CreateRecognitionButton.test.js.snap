// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create recognition component should render correctly 1`] = `
<DocumentFragment>
  .c2 {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  font-size: 27px;
  color: var(--white);
}

.c0 {
  background: var(--white) 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 6px var(--box-shadow-color);
  border-radius: 50%;
  width: 60px;
  height: 60px;
  position: relative;
}

.c1 {
  background: transparent url(ccc) 97% 50% no-repeat padding-box;
  box-shadow: 0px 3px 6px #0000004e;
  width: 45px;
  height: 45px;
  background-color: var(--rust);
  border: none;
  border-radius: 50%;
}

<div
    class="c0 m-auto justify-content-center"
  >
    <div
      class="c1 m-auto text-center"
      id="create-recognition-button"
    >
      <div
        class="c2"
        data-testid="plusSignText"
        font-size="27px"
      >
        +
      </div>
    </div>
  </div>
</DocumentFragment>
`;
